Src/Base/AMReX_DistributionMapping.cpp

```diff
@@ -1196,18 +1197,21 @@ DistributionMapping::SFCProcessorMapDoIt (const BoxArray&          boxes,
 	}
     }
 
-    if (verbose)
+    if (eff || verbose)
     {
         Real sum_wgt = 0, max_wgt = 0;
         for (int i = 0; i < nteams; ++i)
         {
             const long W = LIpairV[i].first;
-            if (W > max_wgt)
-                max_wgt = W;
+            if (W > max_wgt) max_wgt = W;
             sum_wgt += W;
         }
+        *eff = (sum_wgt/(nteams*max_wgt));
```

Comment:
I think you need to check if `eff` is `nullptr` here before writing to the loction.

Reply:
Thanks, I fixed the logic here.